Lib/test/test_typing.py

```diff
@@ -6024,6 +6042,90 @@ def test_get_type_hints(self):
             {'a': typing.Optional[int], 'b': int}
         )
 
+    def test_get_type_hints_generic(self):
+        self.assertEqual(
+            get_type_hints(BarGeneric[int].__origin__),
```

Comment:
The `.__origin__` suffix here looks a bit strange. Would removing it give the same result? If so that would seem more straightforward to me.

Reply:
Related to the discussion [here](https://github.com/python/cpython/pull/27663#discussion_r861978384)